### PR TITLE
Support collect ZGC memory pool metrics

### PR DESCRIPTION
**Pull request by Z-Beatles**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
So, this relies on the protocol update.
Java 9.1 can't release before OAP 9.7, right?

And do we expect UI setting change?

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/memorypool/ZGCCollectorModule.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        for (MemoryPoolMXBean bean : beans) {
+            String name = bean.getName();
+            PoolType type;
+            if (name.equals("ZHeap")) {
+                type = PoolType.ZHEAP_USAGE;
+            } else if (name.equals("Metaspace")) {
+                type = PoolType.METASPACE_USAGE;
+            } else if (name.equals("Compressed Class Space")) {
+                type = PoolType.COMPRESSED_CLASS_SPACE_USAGE;
+            } else if (name.equals("CodeHeap 'non-nmethods'")) {
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/memorypool/ZGCCollectorModule.java:49]: Is the single quote from official definition? It is strange to read.

**Comment by Z-Beatles** [comment]
> So, this relies on the protocol update. Java 9.1 can't release before OAP 9.7, right?
> 
> And do we expect UI setting change?



> So, this relies on the protocol update. Java 9.1 can't release before OAP 9.7, right?
> 
> And do we expect UI setting change?

yes, i will pull a new request on main repo later for UI setting change.

**Comment by wu-sheng** [comment]
Thread pool data is meter oriented.

https://github.com/apache/skywalking/blob/master/oap-server/server-starter/src/main/resources/meter-analyzer-config/threadpool.yaml

I can see JVM metrics don't include pool metric.

https://github.com/apache/skywalking/blob/master/oap-server/server-starter/src/main/resources/oal/java-agent.oal

Could you share which part of OAP would be relative?

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/memorypool/ZGCCollectorModule.java (hunk 1 above)
Comment (Z-Beatles) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/memorypool/ZGCCollectorModule.java:49]: Yes, as we can see i added a panel `JVM Memory Detail (MB)` on UI to show memory pool detail  after `JVM Memory (MB)`.
image

**Review by wu-sheng — DISMISSED** [review]

File: CHANGES.md [hunk 2]
```diff
@@ -15,6 +15,8 @@ Release Notes.
 * Support report MongoDB instance info in Mongodb 4.x plugin.
 * To compatible upper and lower case Oracle TNS url parse.
 * Fix config length limitation.
+* Support collect ZGC memory pool metrics.
```
Comment (wu-sheng) [inline CHANGES.md:18]:
```suggestion
* Support collecting ZGC memory pool metrics. Require OAP 9.7.0 to support these new metrics.
```

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**